chips/nrf52/src/chip.rs

```diff
@@ -52,12 +185,14 @@ impl<I: InterruptService> kernel::Chip for NRF52<I> {
         unsafe {
             loop {
                 if let Some(task) = deferred_call::DeferredCall::next_pending() {
-                    match task {
-                        DeferredCallTask::Nvmc => nvmc::NVMC.handle_interrupt(),
+                    match self.interrupt_service.service_deferred_call(task) {
+                        true => {}
+                        false => panic!("unhandled deferred call task"),
                     }
```

Comment:
Matching over a boolean to only handle the `false` case seems overkill. Why not using `if !self.interrupt_service.service_deferred_call(task) { panic!(...); }`?

Reply:
Still unresolved.

Reply:
Why was this comment marked as resolved? I still don't see the code that fixes that, nor any alternative explanation.

Reply:
oh, sorry, I fixed the instance below where I did the same for the `service_interrupt` call and did not realize I had made the error twice. Actually fixed now.